crates/bevy_pbr/src/environment_map/filter.wgsl

```diff
@@ -0,0 +1,187 @@
+                default { up_vector = vec3(0.0, 0.0, 1.0); }
+            }
+            let frame_x = normalize(cross(up_vector, frame_z));
+            let frame_y = cross(frame_z, frame_x);
+
+            var nx = dir[other_axis0];
+            var ny = dir[other_axis1];
+            let nz = adir[axis];
+
+            let nmax_xy = max(abs(ny), abs(nx));
+            nx /= nmax_xy;
+            ny /= nmax_xy;
+
+            var theta = 0.0;
+            if ny < nx {
+                if ny <= -0.999 { theta = nx; } else { theta = ny; }
+            } else {
+                if ny >= 0.999 { theta = -nx; } else { theta = -ny; }
+            }
+
+            var phi = 0.0;
+            if nz <= -0.999 {
+                phi = -nmax_xy;
+            } else if nz >= 0.999 {
+                phi = nmax_xy;
+            } else {
+                phi = nz;
+            }
+            let theta2 = theta * theta;
+            let phi2 = phi * phi;
+
+            for (var i_super_tap = 0u; i_super_tap < 8u; i_super_tap++) {
+                let index = 8u * axis + i_super_tap;
+                var coeffs_dir0 = array(vec4(0.0), vec4(0.0), vec4(0.0));
+                var coeffs_dir1 = array(vec4(0.0), vec4(0.0), vec4(0.0));
+                var coeffs_dir2 = array(vec4(0.0), vec4(0.0), vec4(0.0));
+                var coeffs_level = array(vec4(0.0), vec4(0.0), vec4(0.0));
+                var coeffs_weight = array(vec4(0.0), vec4(0.0), vec4(0.0));
+
+                for (var i_coeff = 0u; i_coeff < 3u; i_coeff++) {
+                    coeffs_dir0[i_coeff] = coeffs[level][0u][i_coeff][index];
+                    coeffs_dir1[i_coeff] = coeffs[level][1u][i_coeff][index];
+                    coeffs_dir2[i_coeff] = coeffs[level][2u][i_coeff][index];
+                    coeffs_level[i_coeff] = coeffs[level][3u][i_coeff][index];
+                    coeffs_weight[i_coeff] = coeffs[level][4u][i_coeff][index];
+                }
+
+                for (var i_sub_tap = 0u; i_sub_tap < 4u; i_sub_tap++) {
+                    var sample_dir = frame_x * (coeffs_dir0[0u][i_sub_tap] + coeffs_dir0[1u][i_sub_tap] * theta2 + coeffs_dir0[2u][i_sub_tap] * phi2) + frame_y * (coeffs_dir1[0u][i_sub_tap] + coeffs_dir1[1u][i_sub_tap] * theta2 + coeffs_dir1[2u][i_sub_tap] * phi2) + frame_z * (coeffs_dir2[0u][i_sub_tap] + coeffs_dir2[1u][i_sub_tap] * theta2 + coeffs_dir2[2u][i_sub_tap] * phi2);
+
+                    var sample_level = coeffs_level[0u][i_sub_tap] + coeffs_level[1u][i_sub_tap] * theta2 + coeffs_level[2u][i_sub_tap] * phi2;
+
+                    var sample_weight = coeffs_weight[0u][i_sub_tap] + coeffs_weight[1u][i_sub_tap] * theta2 + coeffs_weight[2u][i_sub_tap] * phi2;
+                    sample_weight *= frame_weight;
+
+                    sample_dir /= max(abs(sample_dir[0u]), max(abs(sample_dir[1u]), abs(sample_dir[2u])));
+                    sample_level += 0.75 * log2(dot(sample_dir, sample_dir));
```

Comment:
Why 0.75? I didn't see that in the glTF IBL Sampler from an initial glance.

Reply:
It's from the original shader https://www.activision.com/cdn/research/filter_using_table_128.txt. I didn't use glTF IBL sampler as a reference for anything but diffuse iirc.